src/runtime/virtcontainers/clh.go

```diff
@@ -476,12 +476,21 @@ func (clh *cloudHypervisor) CreateVM(ctx context.Context, id string, network Net
 	// Create the VM config via the constructor to ensure default values are properly assigned
 	clh.vmconfig = *chclient.NewVmConfig(*chclient.NewPayloadConfig())
 
-	// Make sure the kernel path is valid
-	kernelPath, err := clh.config.KernelAssetPath()
+	// Make sure the igvm path is valid
+	igvmPath, err := clh.config.IgvmAssetPath()
 	if err != nil {
 		return err
 	}
-	clh.vmconfig.Payload.SetKernel(kernelPath)
+	clh.vmconfig.Payload.SetIgvm(igvmPath)
```

Comment:
IGVM will only be used in if clh.config.ConfidentialGuest case. We may think of improving the code a bit if we still have time: e.g. use/populate IGVM path only in case clh.config.ConfidentialGuest is true

Reply:
Updated so that igvm must be set if confidential_guest is set, but confidential_guest does not have to be set to use igvm